### PR TITLE
WS-2698: Add property and version num

### PR DESCRIPTION
**Pull request by matt-l-01**
Adding property for our custom artifact httpclient-osgi. Currently used in rosapi1.5 and the java binding.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -121,6 +121,7 @@
         <bt-hamcrest-version>2.2</bt-hamcrest-version>
         <bt-httpasyncclient-version>4.1.5</bt-httpasyncclient-version>
         <bt-httpclient-version>4.5.13</bt-httpclient-version>
+        <bt-custom-httpclient-osgi-version>4.5.13-1</bt-custom-httpclient-osgi-version>
```
Comment (Katsuya-Tomioka) [inline pom.xml:124]: Let's remove the original httpclient-osgi, and replace it with the fork in dependencyManagement section.

Comment (Katsuya-Tomioka) [reply pom.xml:124]: line 814.

Comment (matt-l-01) [reply pom.xml:124]: Good catch. Didn't see that one

**Comment by matt-l-01** [comment]
@Katsuya-Tomioka Any other changes you want to see?

File: pom.xml [hunk 2]
```diff
@@ -809,9 +810,9 @@
                 <version>${bt-httpclient-version}</version>
             </dependency>
             <dependency>
-                <groupId>org.apache.httpcomponents</groupId>
+                <groupId>com.basistech.org.apache.httpcomponents</groupId>
```
Comment (Katsuya-Tomioka) [inline pom.xml:813]: :100: 
Looks good to me.

**Comment by Katsuya-Tomioka** [comment]
@dscorbett @juliencherry 
We want to swap httpcomonents-client-osgi with our fork to patch comons-codec vuln. P&C is the only consumer of this artifact, so we could remove it from open-source-parent, except the version needs to align with httpcomponent's -- it's good as is, I think. Non-OSGi client doesn't need to patch as long as commons-codec is overridden.

**Comment by dscorbett** [comment]
Adding a new property (`bt-custom-httpclient-osgi-version`) and adding a new dependency (`com.basistech.org.apache.httpcomponents:httpclient-osgi`) are API additions, which deserve minor version bumps. Removing an existing dependency (`org.apache.httpcomponents:httpclient-osgi`) is a backwards-incompatible change, because it will break any child POM that uses that dependency without specifying the version itself, so it deserves a major version bump. Therefore, if you merge and release this, call it version 14.0.0.

If you also release an internal parent POM, the same logic applies, so you should theoretically bump it from 69.0.0 to 70.0.0. However, WS-2698 was opened after we had already started using parent POM 69.0.0. It would cause a lot of churn to call it 70.0.0, since we would have to rerelease everything. The most expedient solution, if a new internal parent POM is necessary, is to call it 69.0.1. As long you are certain that nothing we’ve released in the past week will be affected by this change, it won’t actually break anything, though it is theoretically unsafe.

**Pull request merged**